tests/test_html.py

```diff
@@ -0,0 +1,437 @@
+
+
+def test_to_html_preserves_line_breaks_for_textbox_large_questions(content_summary):
+    question = content_summary.get_question("myMultipleLines")
+    assert to_html(question) == Markup(
+        "Text with multiple lines.<br><br>This is the second line."
+    )
+
+
+@pytest.mark.parametrize(
+    "question, expected",
+    [
+        ("myLowercaseText", "Text to be capitalised"),
+        ("myLowercaseTextarea", "Text to be capitalised<br>with a line break."),
+        (
+            "myLowercaseList",
+            """<ul class="govuk-list govuk-list--bullet">\n  <li>Line 1</li>\n  <li>Line 2</li>\n</ul>""",
+        ),
+    ],
+)
+def test_to_html_can_capitalize_first(content_summary, question, expected):
+    question = content_summary.get_question(question)
+    assert to_html(question, capitalize_first=True) == expected
+
+
+@pytest.mark.parametrize(
+    "question, expected",
+    [
+        (
+            "myLinkText",
+            """<a href="https://www.gov.uk" class="app-break-link" rel="external">https://www.gov.uk</a>""",
+        ),
+        (
+            "myLinkTextarea",
+            """Here is a URL:<br><br>"""
+            """<a href="https://www.gov.uk" class="app-break-link" rel="external">https://www.gov.uk</a>""",
+        ),
+        (
+            "myLinkList",
+            dedent(
+                """\
+        <ul class="govuk-list govuk-list--bullet">
+          <li><a href="https://www.gov.uk" class="app-break-link" rel="external">https://www.gov.uk</a></li>
+          <li><a href="https://www.gov.uk/" class="app-break-link" rel="external">https://www.gov.uk/</a></li>
+        </ul>"""
+            ),
+        ),
+    ],
+)
+def test_to_html_can_format_links(content_summary, question, expected):
+    question = content_summary.get_question(question)
+    assert to_html(question, format_links=True) == expected
+
+
+@pytest.mark.parametrize(
+    "question, expected",
+    (
+        ("myUpload", """<a class="govuk-link" href="#">Upload Question</a>"""),
+        ("myScriptInjection", "&lt;script&gt;do something bad&lt;/script&gt;"),
+        (
+            "mySimpleList",
+            dedent(
+                """\
+        <ul class="govuk-list govuk-list--bullet">
+          <li>Hello</li>
+          <li>World</li>
+        </ul>"""
+            ),
+        ),
+    ),
+)
+@pytest.mark.parametrize("autoescape", (True, False))
+def test_to_html_returns_html_which_is_safe_for_jinja(
+    content_summary, question, expected, autoescape
+):
+    question = content_summary.get_question(question)
+    html = to_html(question)
+
+    env = jinja2.Environment(autoescape=autoescape)
+    template = env.from_string("{{ html }}")
+
+    assert template.render(html=html) == expected
+
+
+def test_to_summary_list_rows(content_summary):
+    questions = content_summary.sections[0].questions
+    summary_list_rows = to_summary_list_rows(questions)
+
+    assert summary_list_rows
+    assert len(summary_list_rows)
```

Comment:
We may as well be explicit here - `assert len(summary_list_rows) == 22` (and then we can remove the assertion above it).